lib/aws.ts

```diff
@@ -43,65 +61,75 @@ export interface Account {
  * AWS client using the AWS SDK for JS with no special configuration
  */
 export class DefaultAwsClient implements IAws {
-  private readonly AWS: typeof import('aws-sdk');
   private account?: Account;
+  private config: Configuration;
 
-  constructor(profile?: string) {
-    // Force AWS SDK to look in ~/.aws/credentials and potentially use the configured profile.
-    process.env.AWS_SDK_LOAD_CONFIG = '1';
-    process.env.AWS_STS_REGIONAL_ENDPOINTS = 'regional';
-    process.env.AWS_NODEJS_CONNECTION_REUSE_ENABLED = '1';
-    if (profile) {
-      process.env.AWS_PROFILE = profile;
+  constructor(private readonly profile?: string) {
+    process.env.AWS_PROFILE = profile;
+    const clientConfig: STSClientConfig = {
+      customUserAgent: USER_AGENT,
     }
-    // Stop SDKv2 from displaying a warning for now. We are aware and will migrate at some point,
-    // our customer don't need to be bothered with this.
-    process.env.AWS_SDK_JS_SUPPRESS_MAINTENANCE_MODE_MESSAGE = '1';
+    this.config = {
+      clientConfig,
+      credentials: fromNodeProviderChain({
+        profile: this.profile,
+        clientConfig,
+      }),
+    };
+  }
 
-    // We need to set the environment before we load this library for the first time.
-    // eslint-disable-next-line @typescript-eslint/no-require-imports
-    this.AWS = require('aws-sdk');
+  public async s3Client(options: ClientOptions): Promise<S3Client> {
+    return new S3Client(await this.awsOptions(options));
   }
 
-  public async s3Client(options: ClientOptions) {
-    return new this.AWS.S3(await this.awsOptions(options));
+  public async upload(params: PutObjectCommandInput, options: ClientOptions): Promise<CompleteMultipartUploadCommandOutput> {
+    try {
+      const upload = new Upload({
+        client: await this.s3Client(options),
+        params,
+      });
+
+      return upload.done();
+    } catch (e) {
+      // TODO: add something more useful here
```

Comment:
todo?

Reply:
I'll add something here before I remove the release candidate tag. I need to actually test the failure modes in the CLI to figure out what will be useful information here.